src/backend/vulkan/src/physical_device.rs

```diff
@@ -1083,6 +1156,63 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
         }
     }
 
+    fn query_external_buffer_properties(
+        &self,
+        usage: hal::buffer::Usage,
+        sparse: hal::memory::SparseFlags,
+        memory_type: hal::external_memory::ExternalMemoryType,
+    ) -> Result<
+        hal::external_memory::ExternalBufferProperties,
+        hal::external_memory::ExternalMemoryQueryError,
+    > {
+        let external_memory_capabilities_extension = match &self
+            .instance
+            .external_memory_capabilities
+        {
+            Some(functor) => functor,
+            _ => return Err(hal::external_memory::ExternalMemoryQueryError::UnsupportedFeature),
```

Comment:
I don't think this is necessary. We should just `unwrap()`.
This is a very low-level API. The only kind of errors we should be returning are ones that users can't expect.
In case of this, the user knows when they requested the `EXTERNAL_MEMORY` feature, hence they know when they can call this.

Reply:
Ok, so i will apply the change for every function returning `UnsupportedFeature`.

Reply:
Done.